website/docs/getting-started/template-walkthrough.md

```diff
@@ -0,0 +1,146 @@
+import { Function, StepFunction } from "functionless";
+
+const app = new App();
+
+const stack = new Stack(app, "MyStack");
+
+const sayFunction = new Function(
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    console.log(event.message);
+    return;
+  }
+);
+
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+We start by importing constructs from `aws-cdk-lib` and `functionless`:
+
+```typescript
+import { App, Stack } from "aws-cdk-lib";
+import { Function, StepFunction } from "functionless";
+```
+
+Functionless exposes AWS CDK compatible constructs from its top level API.
+In this example, we're using the `Function` and `StepFunction` constructs which map to the AWS Lambda and AWS Step Functions services.
+
+Next, we initialize `app` which will be the root of our CDK construct tree:
+
+```typescript
+const app = new App();
+```
+
+CDK models infrastructure as a tree of constructs with a single instance of `App` at the root.
+To add constructs to the tree, we pass the parent construct as the first argument when instantiating a child construct.
+
+This is exactly what we do next with `Stack`:
+
+```typescript
+const stack = new Stack(app, "MyStack");
+```
+
+`stack` will be the parent of the rest of the constructs we create.
+In CDK most resources are attached to a `Stack` instance as they are the smallest deployable unit of infrastructure.
+
+An `App` may contain many stacks, but that is outside the scope of this document.
+You can read more about [`App`][cdk-app] and [`Stack`][cdk-stack] in the CDK documentation.
+
+Once we have an instance of `stack`, we're ready to attach our first `Function`:
+
+```typescript
+const sayFunction = new Function(
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    // This code will run on AWS Lambda
+    console.log(event.message);
+    return;
+  }
+);
+```
+
+This might not be the most interesting code in the world, but it offers us our first glimpse at the magic of Functionless.
+
+If you've used CDK before, you're probably familiar with the seperation it imposes between infrastructure and application code.
+Functionless bridges this gap by letting you to write application code inline with the infrastructure it sits atop.
+
+This enables a whole new level of interaction between your application and its infrastructure.
+
+We can see this in action the `StepFunction` construct definition:
+
+```typescript
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  // This will run in an AWS Step Function
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+There is a lot to unpack in these three lines.
+
+First off, we're able to call the lambda `sayFunction` construct we just defined as if it were a regular nodejs function.
+This function uses the correct types inferred from the closure used in the function definition.
```

Comment:
Is the word "closure" too technical? Maybe just call it "function". Also, maybe re-show that piece of the code here for context.

Reply:
Yeah it probably is, will revise.

Reply:
changed to "arrow function" which should be more familiar to TypeScript devs.